eBCSgen/Parsing/ParseBCSL.py

```diff
@@ -212,10 +247,23 @@ def regulation_def(self, matches):
         return matches[0]
 
     def regular(self, matches):
-        re = "".join(matches[1:])
-        # might raise exception
-        regex.compile(re)
+        re = self.tree_to_regex_string(matches[1])
+        try:
+            regex.compile(re)
+        except regex.error as e:
+            raise RegulationParsingError(f"Invalid regular expression: {re}. Error: {e}")
         return Regular(re)
+    
+    def tree_to_regex_string(self, tree):
+        regex_string = ""
+    
+        if isinstance(tree, Token):
+                return tree.value
+        elif isinstance(tree, Tree):
+            for child in tree.children:
+                regex_string += self.tree_to_regex_string(child)
+                
+        return regex_string
```

Comment:
I feel like there is exactly the same function in [`Rate` - `tree_to_string`](https://github.com/mopichalova/eBCSgen/blob/bb8edf5722abd907364d3636f440f430ef0be5a0/eBCSgen/Core/Rate.py#L256) - perhaps use that one or try to generalise them if possible (maybe use better location for it).

Reply:
Right, I missed that, thank you! I moved the `tree_to_string` function to new file `utils.py` in `eBCSgen` as it is used in `eBCSgen/Core` in `Rate.py` and `Formula.py` and also in `eBCSgen/Parsing/ParseBCSL.py`.